### PR TITLE
Refactor WordBoard Initialization and Attributes

### PR DESCRIPTION
**Pull request by yilmazcabuk**
## Description
This pull request refactors the `WordBoard` class initialization process and modifies its attributes for improved readability and maintainability. The changes also introduce dictionaries for various attributes and update the calculation of board attributes based on the new configuration.

## Changes
- Modified the initialization process of the `WordBoard` class for improved clarity.
- Changed the `words` attribute to be a set instead of a list.
- Introduced dictionaries for `letter_values,` `letter_multipliers,` `word_multipliers`, and `board_value`.
- Updated `word_values` to be a list of tuples containing word values and words.
- Renamed the parameter of the `set_board` method to `game_board`.
- Recalculated various attributes based on the updated board configuration.

## Motivation
The motivation behind these changes is to enhance the organization and maintainability of the `WordBoard` class. By introducing dictionaries and renaming attributes, we aim to make the code more understandable and easier to extend in the future.

### CODE REVIEW

File: spellcast.py [hunk 1]
```diff
@@ -88,6 +97,19 @@ def __init__(self):
...
+        self.letter_values = {}
+        self.letter_multipliers = defaultdict(lambda: 1)
+        self.word_multipliers = defaultdict(lambda: 1)
+        self.board = []
+        self.row_count = 0
+        self.column_count = 0
+        self.total_count = Counter()
+        self.board_value = {}
+        self.word_values = []
+        self.calculate_value = None
```
Comment (vscala) [inline spellcast.py]: It looks like `calculate_value` is only used in the recalculate function so it probably doesn't need to be a attribute of WordBoard.

File: spellcast.py [hunk 2]
```diff
@@ -88,6 +97,19 @@ def __init__(self):
...
         """
+        self.words = set()
+        self.letter_values = {}
+        self.letter_multipliers = defaultdict(lambda: 1)
+        self.word_multipliers = defaultdict(lambda: 1)
+        self.board = []
+        self.row_count = 0
+        self.column_count = 0
+        self.total_count = Counter()
+        self.board_value = {}
```
Comment (vscala) [inline spellcast.py]: It looks like `board_value` is never actually used, it can probably be removed.

File: spellcast.py [hunk 3]
```diff
@@ -88,6 +97,19 @@ def __init__(self):
         Reads words from a file and initializes various attributes needed for
         managing the game board.
         """
+        self.words = set()
+        self.letter_values = {}
```
Comment (vscala) [inline spellcast.py]: `letter_values` can probably be moved to a local scope in recalculate.

File: spellcast.py [hunk 4]
```diff
@@ -88,6 +97,19 @@ def __init__(self):
         Reads words from a file and initializes various attributes needed for
         managing the game board.
         """
+        self.words = set()
```
Comment (vscala) [inline spellcast.py]: `words` is being initialized here as a set but becomes a list on line 114, consider making this a list or changing line 114 to  a set.

File: spellcast.py (hunk 1 above)
Comment (yilmazcabuk) [reply spellcast.py]: Done.

File: spellcast.py (hunk 2 above)
Comment (yilmazcabuk) [reply spellcast.py]: Removed.

File: spellcast.py (hunk 4 above)
Comment (yilmazcabuk) [reply spellcast.py]: Fixed and renamed.

File: spellcast.py (hunk 3 above)
Comment (yilmazcabuk) [reply spellcast.py]: Scoped.

File: spellcast.py [hunk 5]
```diff
@@ -130,35 +124,32 @@ def recalculate(self):
...
-            for character in word
-            if character.lower() in LETTERS_AND_VALUES
-        ) + (
-            long_word_bonus_points if len(word) > long_word_minimum_letter_count else 0
-        )
-
-        self.word_values = [(self.calculate_value(word), word) for word in self.words]
+
+        def calculate_value(word):
+            sum(
```
Comment (vscala) [inline spellcast.py]: This is missing `return`

**Review by vscala — CHANGES_REQUESTED** [review]

File: spellcast.py (hunk 5 above)
Comment (yilmazcabuk) [reply spellcast.py]: It was actually working as is but still the suggested change was applied.

**Review by vscala — APPROVED** [review]

**Pull request merged**